src/printer.js

```diff
@@ -2106,11 +2122,29 @@ function printArgumentsList(path, options, print) {
     const shouldBreak = shouldGroupFirst
       ? printed.slice(1).some(willBreak)
       : printed.slice(0, -1).some(willBreak);
+
+    let printedLastArgExpanded;
+    if (shouldGroupFirst) {
+      printedLastArgExpanded = printed;
+    } else {
+      // We want to print the last argument with a special flag
+      let i = 0;
+      path.each(function(argPath) {
+        if (i++ === args.length - 1) {
+          printedLastArgExpanded = printed
+            .slice(0, -1)
+            .concat(path.call(
+              print.withArgs({expandLastArg: true})
```

Comment:
I don't understand why you need `withArgs`. Why not do `path.call(p => print(p, { expandLastArg: true })))`? Am I missing what's wrong there?